### PR TITLE
Change ReadTheDocs to use conda and not fail

### PR DESCRIPTION
**Pull request by Cadair**
For the last little while all our RTD builds have been stalling, so the documentation has not been updating.

This PR fixes this.

Highlights:

* RTD build now uses conda, meaning we have access to all packages.
* Some lightcurve data is now in `sunpy.data.sample`
* The `LYRALightCurve` plot now uses level 3 data. This, as it turns out, was the problem, we were exceeding the memory limits by trying to plot the full 70Mb dataset.

@ehsteve can you update your sample data mirrors please.

### CODE REVIEW

**Comment by Cadair** [comment]
ok, this seems to have passed on my RTD fork.

Review please!

File: doc/source/conf.py [hunk 1]
```diff
@@ -34,13 +34,8 @@
 
 modules = {}
 
-try:
-    import skimage
-except ImportError:
-    from mock import Mock
-    mock = Mock()
-    modules.update({'skimage':mock, 'skimage.feature':mock.module})
-
```
Comment (dpshelio) [inline doc/source/conf.py]: Ahhh!! that's the mock for :)

**Comment by dpshelio** [comment]
:+1:

**Comment by Cadair** [comment]
<rant>
Having the documentation build connect to the internet and download data is pain. Not only does it massively slow down the build when you are writing documentation it causes fundamental problems.
Firstly, it is causing the success or faliure of our travis and RTD build to depend on the existance of online services, which is exactly the problem we have with the 'online' test build, and why we put so much effort into making that build optional.

Secondly, and most importantly, one of the key set of *users* who might want to locally build the documentation on their machines are those with weak internet connections. Having the actual *build* process of the documentation rely on that internet connection seems to defeat the idea of having the ability for users to build their own docs.

For these reasons, this PR is about to remove all `.. plot::` directives that connect to the internet, unless anyone knows how to make those somehow optional.

**Comment by dpshelio** [comment]
What about the vso, helioviewer and hek ones? 

If the `.. plot::` would be optional... wouldn't be needed anyway when building the docs for RTD?

**Comment by Cadair** [comment]
@dpshelio The only ones I have found which I have had to remove are the helioviewer ones.

File: doc/source/guide/data_types/spectra.rst [hunk 2]
```diff
@@ -97,7 +97,7 @@ To get rid of the noise, we could also clip low intensities by setting vmin.
 
 .. plot::
 
-    from matplotlib import pyplot as plt
+    import matplotlib.pyplot as plt
     import sunpy
     import sunpy.data.sample
     from sunpy.spectra.sources.callisto import CallistoSpectrogram
```
Comment (Cadair) [inline doc/source/guide/data_types/spectra.rst]: @dpshelio @segfaulthunter You have any idea why this example would be trying to access the internet? At least I think it's this one.

Comment (dpshelio) [inline doc/source/guide/data_types/spectra.rst]: If you ask for data, then yes... and if I remember right, in the documents it does a "expand" so it download the file before or after the main one...

Comment (Cadair) [inline doc/source/guide/data_types/spectra.rst]: where is this example asking for data?

Comment (dpshelio) [inline doc/source/guide/data_types/spectra.rst]: well... a few lines below uses `BIR_20110922_101500_01.fit'` and another one, so they have to come from the net... also the `from_range()` calls the files and the extent which I don't see it in here...

Comment (Cadair) [inline doc/source/guide/data_types/spectra.rst]: that's not within the `plot` though right?

Comment (dpshelio) [inline doc/source/guide/data_types/spectra.rst]: nope... and I don't think `plot()` calls anything online

Comment (dpshelio) [inline doc/source/guide/data_types/spectra.rst]: but the one in [line 32](#diff-41807c18f317e1bc6a47eaca781516a3R32) should do

Comment (Cadair) [inline doc/source/guide/data_types/spectra.rst]: only the lines inside the `.. plot::` blocks should be executed.

Comment (dpshelio) [inline doc/source/guide/data_types/spectra.rst]: and the line 42 is inside `.. plot::`

Comment (Cadair) [inline doc/source/guide/data_types/spectra.rst]: ah yes, my bad. Hmm looks like I might have to re-write that example.

**Comment by Cadair** [comment]
Ok @sunpy/sunpy-developers This PR needs your input.

I have brutally removed all the sphinx `.. plot::` directives that require an internet connections, for reasons ranted above. I have not yet replaced these plots with static images, just code examples (this can change, but I wanted to give some time for discussion).

Comments please.

**Comment by wafels** [comment]
I agree that you shouldn't need to connect to helioviewer to build the sunpy docs.  But I think it might be worthwhile to include PNG files of the results of the helioviewer requests.  But where would that go in SunPy - in the sample data?

**Comment by Cadair** [comment]
@wafels they could just go in the doc source.

**Comment by Cadair** [comment]
@wafels I added back the helioviewer images, you good with this PR now?

**Comment by wafels** [comment]
+1

**Pull request merged**